src/main/java/com/laserfiche/api/client/httphandlers/UsernamePasswordHandler.java

```diff
@@ -0,0 +1,74 @@
+package com.laserfiche.api.client.httphandlers;
+
+import com.laserfiche.api.client.model.CreateConnectionRequest;
+import com.laserfiche.api.client.model.SessionKeyInfo;
+import com.laserfiche.api.client.selfhosted.TokenClient;
+import com.laserfiche.api.client.selfhosted.TokenClientImpl;
+
+import java.util.concurrent.CompletableFuture;
+
+public class UsernamePasswordHandler implements HttpRequestHandler {
+    private String accessToken;
+    private final String GRANTTYPE = "password";
+    private String repositoryId;
+    private String baseUrl;
+    private TokenClient client;
+    private CreateConnectionRequest request;
+
+    /**
+     * Creates a username and password authorization handler for self hosted API server
+     *
+     * @param repositoryId Repository name
+     * @param username     The username used with "password" grant type.
+     * @param password     The password used with "password" grant type.
+     * @param baseUrl      APIServer Base Url e.g. https://example.com/LFRepositoryAPI
+     * @param client       OPTIONAL
+     */
+    public UsernamePasswordHandler(String repositoryId, String username, String password, String baseUrl,
+            TokenClient client) {
+        this.baseUrl = baseUrl.endsWith("LFRepositoryAPI") ? baseUrl : baseUrl.substring(0, baseUrl.lastIndexOf("/"));
```

Comment:
better to not hardcode `"LFRepositoryAPI"` Would this work?
`baseUrl.endsWith("/") ? baseUrl.substring(0, baseUrl.lastIndexOf("/")) : baseUrl;`

Reply:
well i was thinking about an additional edge like this for example `https://camrose.laserfiche.com/LFRepositoryAPI/test/testsefsdfsd` which is why i hardcoded that

Reply:
done